### PR TITLE
feat: basic addon surpport stop/start

### PR DESCRIPTION
**Pull request by wangzhuzhen**
#### What this PR does / why we need it:

when erda deployment deleted, the addon pods will always running unless you delete addon,but addon may be reused for next deployment, so to release resources (cpu/memorys), can stop addon by scale addon services replicas to 0, this can retain addon services's PVC for reused，If you want reuse addon adgin,  can start addon by scale addon services replicas  original values 

#### Specified Reviewers:

/assign @sixther-dc @luobily @iutx 

#### ChangeLog
Added an API for scale down or scale up addon

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     Added an API for scale down or scale up addon    |
| 🇨🇳 中文    |        addon scale API 支持停止/启动 addon      |

### CODE REVIEW

File: modules/orchestrator/services/addon/addon.go [hunk 1]
```diff
@@ -1631,6 +1629,318 @@ func (a *Addon) Delete(userID, routingInstanceID string) error {
 	return nil
 }
 
+// Scale 根据 routingInstanceID 调整 addon 实例副本数实现:
+// 停止: 副本数设置为 0
+// 启动: 副本数调整为期望值
+// 平台级 addon 只支持扩，不支持缩
```
[the remark this answers is not in the document]
Comment (wangzhuzhen) [reply modules/orchestrator/services/addon/addon.go:1635]: Only basic addon can scale, scale logic will detect addon type ensure to support scale action.

File: apistructs/addon.go [hunk 2]
```diff
@@ -553,6 +553,8 @@ type AddonFetchResponseData struct {
 	CustomAddonType string `json:"customAddonType"`
 	// TenantOwner addon 租户owner的 instancerouting id
 	TenantOwner string `json:"tenantOwner"`
+	// IsInsideAddon addon 是否是 inside addon（如 kafka addon 中的 inside addon 是 zookeeper addon）
+	IsInsideAddon string `json:"isInsideAddon`
```
[the remark this answers is not in the document]
Comment (wangzhuzhen) [reply apistructs/addon.go:557]: In some table Inside_addon use string not bool.

File: modules/orchestrator/services/addon/addon.go [hunk 3]
```diff
@@ -1631,6 +1629,318 @@ func (a *Addon) Delete(userID, routingInstanceID string) error {
...
+	}
+
+	// 根据 addon 状态，判断当前是否支持对应 action 的操作
+	err = addonCanScale(routingIns.AddonName, routingIns.RealInstance, routingIns.Plan, routingIns.Version, routingIns.Status, action)
+	if err != nil {
+		logrus.Errorf("%v", err)
+		return err
+	}
+
+	// 若引用关系存在，不能停止
```
[the remark this answers is not in the document]
Comment (wangzhuzhen) [reply modules/orchestrator/services/addon/addon.go:1684]: No, scale logic won't change reference count, if reference count greater than 0, scale down will abort。If runtime stopped means runtime have deleted，then the reference count is equal to 0

File: modules/scheduler/executor/plugins/k8s/addon/redis/legacy/redis.go [hunk 4]
```diff
@@ -261,9 +261,74 @@ func (ro *RedisOperator) Remove(sg *apistructs.ServiceGroup) error {
...
+	if err != nil {
+		return fmt.Errorf("failed to update redisfailover, %s/%s, err: %v, body: %v ", oldRedis.Namespace, oldRedis.Name, err, b.String())
+	}
+	if !resp.IsOK() {
+		return fmt.Errorf("failed to update redisfailover, %s/%s, statuscode: %v, body: %v", oldRedis.Namespace, oldRedis.Name, resp.StatusCode(), b.String())
+	}
+
+	return nil
+
+	//return fmt.Errorf("redisoperator not impl Update yet")
```
Comment (luobily) [inline modules/scheduler/executor/plugins/k8s/addon/redis/legacy/redis.go:331]: delete it.

Comment (wangzhuzhen) [reply modules/scheduler/executor/plugins/k8s/addon/redis/legacy/redis.go]: Resolved.

**Pull request merged**